File: backend/src/Blizka.App/Domain/Entities/Interest.cs
using Blizka.App.Domain.Enums;

namespace Blizka.App.Domain.Entities;

public sealed class Interest
{
    public Guid Id { get; set; }

    public InterestCategory Category { get; set; }

    public string NameRu { get; set; } = string.Empty;

    public string NameBe { get; set; } = string.Empty;

    public string NameEn { get; set; } = string.Empty;

    public bool IsCustom { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public ICollection<UserInterest> UserInterests { get; set; } = [];
}

File: frontend/src/domains/interests/types/interest.ts
/** Сверено с backend: `Blizka.Api/Interests/InterestDtos.cs`. */
export type Interest = {
  id: string
  name: string
  /** Создан пользователями, а не входит в стартовый каталог. */
  isCustom: boolean
}

/** Сверено с backend: `Blizka.App/Domain/Enums/InterestCategory.cs`. */
export type InterestCategory =
  'sport' | 'creativity' | 'entertainment' | 'foodAndDrinks' | 'growthAndTravel' | 'custom'

export type InterestGroup = {
  category: InterestCategory
  interests: Interest[]
}
